CSharp/Concentus/Concentus.csproj

```diff
@@ -21,4 +21,8 @@
     <PackageReference Include="System.Numerics.Vectors" Version="4.5.0" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="runtimes\" />
```

Comment:
Just out of curiosity, are you planning to move away from managed encoding/decoding? Or is the intent just to give people the option to use libopus if they want?

Reply:
Also, that `linux-armv6` directory is actually wrong:

```
$ readelf -h libopus.so | grep ABI
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Flags:                             0x5000400, Version5 EABI, hard-float ABI
$ readelf -d libopus.so | grep arm
 0x00000001 (NEEDED)                     Shared library: [ld-linux-armhf.so.3]
```

`armhf` means Arm v7 (32-bit) with hard float ABI. In .NET parlance, that's simply `linux-arm`.

As a side note, for complete coverage of the platforms that .NET is "typically" run on these days, you'd need `linux-arm`, `linux-arm64`, `linux-x64`, `osx-arm64`, `osx-x64`, `win-arm64`, `win-x64`, `win-x86`.

Reply:
Out of boredom and curiosity, I went and tried to build libopus 1.5.1 for all of those RIDs with `zig cc`. The binaries are available here in case you need them: https://drive.google.com/file/d/1KNbv6s8vFAqbT9goyTLt83oyih1Uvdaz/

They are built for Linux 3.10 (glibc 2.17), macOS 11.7, and Windows 8+. (Basically, I started by checking the oldest supported systems for .NET 6, and then adjusted upwards if Zig didn't support systems that old. This was only the case for macOS, where .NET 6 supports 10.15, but Zig only supports 11.7+.)

If you don't trust these (which is totally fair!), I can let you know the full instructions to cross-compile all of these with `zig cc`.

Reply:
1. Yeah, I plan to move users to a higher-level `OpusCodecFactory.CreateEncoder()` method which will just give them the native P/Invoke version if supported by their platform, or the managed code as a fallback.
2. Interesting; I compiled that .so on a Raspberry Pi 1B+, which is ARMv6 but with hard-float support (as an extension?) so maybe there is some weirdness with the ABI there. My intention was to target RPi1 which does segfault if it tries to run modern dotnet, but can still run this code via Mono.
3. Thanks for the builds! I do have to ask the Opus team how best to license loose binaries like this since I presumably need to tag the exact source they were compiled from, but I'm sure these will come in handy

Reply:
> I'll check if I can build a binary for v6 + hard float with `zig cc`.

Doesn't look like there's an easy way to do this, unfortunately. `zig cc` only speaks v7 (32-bit) and v8 (64-bit) right now. But if you built your `linux-armv6` binary from the `v1.5.1` tag too, then it should be fine anyway.

Reply:
Some additional points for consideration:

* Those binaries are built with fixed point disabled and floating point API enabled, as those are the defaults. Is that an issue? I just noticed that the Concentus README mentions fixed point. :thinking:
* Fast float approximation was enabled for all of them ([also the default](https://github.com/xiph/opus/blob/ab4e83598e7fc8b2ce82dc633a0fc0c452b629aa/configure.ac#L194)).
* I didn't enable any of the [fancy ML stuff from Opus 1.5](https://opus-codec.org/demo/opus-1.5). Some of it (PLC & DRED) produces experimental bitstreams so is probably not appropriate to enable for Concentus at this time, but LACE and NoLACE do not affect the bitstream compatibility, so might be interesting to enable?
* I didn't enable support for custom modes (44.1 kHz and 2^n frames).

Let me know if you think I should re-do the builds with any of these changed.